cedar-policy-validator/src/coreschema.rs

```diff
@@ -307,11 +306,11 @@ pub enum RequestValidationError {
         /// Action which it is not valid for
         action: Arc<ast::EntityUID>,
     },
-    /// Error computing the type of the `Context`; see the contained error type
-    /// for details about the kinds of errors that can occur
-    #[error("context is not valid: {0}")]
+    /// Error computing the type of the `Context`.
+    /// Currently the only possible error is due to an extension function lookup failure.
+    #[error("invalid context: {0}")]
     #[diagnostic(transparent)]
-    TypeOfContext(GetSchemaTypeError),
+    TypeOfContext(ExtensionFunctionLookupError),
```

Comment:
I'm surprised that the other variants of `GetSchemaTypeError` aren't possible here.  Shouldn't most of them be possible to encounter when trying to typecheck a `Context`?

Also, even if they aren't possible today, does it seem possible that future changes could leave us wanting to be able to return the other variants of `GetSchemaTypeError` here?  This change would box us in from being able to return those other variants in the future.

Reply:
What about the heterogeneous-set case?  do the public constructors for `Context` rule out heterogeneous sets?  I would be surprised if they do -- actually I think it's a bug if they do, since in unvalidated Cedar, heterogeneous sets should be allowed

Reply:
Different code paths. The heterogenous set error can only be returned when trying to infer the type of a set. This will never happen for context validation because the schema says exactly what type the set should contain. A heterogeneous set will cause `typecheck_restricted_expr` to return `Ok(false)`. (So no errors, but there's a problem with the type of the context.)

Reply:
Which is also making me wonder: since we are masking other errors in the `typecheck_restricted_expr` function anyways, maybe we should just drop the `Result` type and return false in case of an `ExtensionFunctionLookupError`?

Reply:
Reverted this change in the last commit. I think I can put up a PR later today with some improvements in this part of the code